examples/platform/silabs/SiWx917/SiWxWifiInterface.cpp

```diff
@@ -919,8 +922,7 @@ void wfx_dhcp_got_ipv4(uint32_t ip)
     ChipLogDetail(DeviceLayer, "DHCP OK: IP=%d.%d.%d.%d", wfx_rsi.ip4_addr[0], wfx_rsi.ip4_addr[1], wfx_rsi.ip4_addr[2],
                   wfx_rsi.ip4_addr[3]);
     /* Notify the Connectivity Manager - via the app */
-    wfx_rsi.dev_state |= WFX_RSI_ST_STA_DHCP_DONE;
+    wfx_rsi.dev_state.Set(WifiState::kStationDhcpDone, WifiState::kStationReady);
```

Comment:
I don't think this is correct it would do
`constexpr BitFlags & Set(FlagsEnum flag, bool isSet) `

```suggestion
    wfx_rsi.dev_state.Set(WifiState::kStationDhcpDone).Set(WifiState::kStationReady);
```